src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs

```diff
@@ -1745,14 +1747,104 @@ private void CreatePropertyTypeData()
         }
     }
 
-    private void CreateLanguageData() =>
-        ConditionalInsert(
+    private void CreateLanguageData()
+    {
+        // For languages we support the installation of records that are additional to the default installed data.
+        // We can do this as they are specified by ISO code, which is enough to fully detail them.
+        // All other customizable install data is specified by GUID, and hence we only know about the set that are installed by default.
+        InstallDefaultDataSettings? languageInstallDefaultDataSettings = _installDefaultDataSettings.Get(Constants.Configuration.NamedOptions.InstallDefaultData.Languages);
+        if (languageInstallDefaultDataSettings?.InstallData == InstallDefaultDataOption.Values)
+        {
+            // Insert the specified languages, ensuring the first is marked as default.
+            bool isDefault = true;
+            foreach (var isoCode in languageInstallDefaultDataSettings.Values)
+            {
+                if (!TryCreateCulture(isoCode, out CultureInfo? culture))
+                {
+                    continue;
+                }
+
+                var dto = new LanguageDto
+                {
+                    IsoCode = culture.Name,
+                    CultureName = culture.DisplayName,
+                    IsDefault = isDefault
+                };
+                _database.Insert(Constants.DatabaseSchema.Tables.Language, "id", true, dto);
+                isDefault = false;
+            }
+        }
+        else
+        {
+            // Conditionally insert the default language.
+            string isoCode = "en-US";
+            if (TryCreateCulture(isoCode, out CultureInfo? culture))
```

Comment:
We can now remove the temporary variable here:
```suggestion
            if (TryCreateCulture("en-US", out CultureInfo? culture))
```